### PR TITLE
Test Performance and Review Updates

### LINKED ISSUE #287: Ability to use filtered WFS in webgeom POST
see comments on #277 

From Blodg:
Something to note though, you don't actually have to do the gmlid lookup if you have implemented filtering like this. The GDP can take the filtered WFS request rather than a list of gmlids. You do have to include `<wfs:PropertyName>the_geom</wfs:PropertyName>` for geoserver-shapefile WFS and possibly other names for the geometry column for other WFS implementations, but you could skip the gmlid lookup part.

### LINKED ISSUE #359: make tests run faster
Let's collect ideas of how to make the tests run faster here then implement them. Quickly. 

The big one I see if making all the prism calls 1 time step by adjusting the end time.

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
Some updates in here addressing comment on #362 and #361. 

I modified the filter that gets sent in an execute request to not use gmlids. This should be an improvement because we don't have to query the WFS for all available values to filter. Fixes #287

The testing updates make tests run a good bit faster, especially when the network is slow. I mostly moved tests around and made local .rds artifacts where it wasn't necessary to call out to the server to initialize objects etc. There are still some tests that do things that are needlessly chatty or touch more data than they need to, I left a few comments to that affect. Happy to revert any and all the test changes, but I think they are a step in the right direction to test the geoknife code rather than the web services. This should fix #359.

### CODE REVIEW

File: R/geoknife-generic.R [hunk 1]
```diff
@@ -117,7 +117,7 @@ parseXMLgeoms	<-	function(xml){
   # ignore namespaces
   xpath <- sprintf("//*[local-name()='%s']/*[local-name()='%s']/*[local-name()='%s']",parentKey,childKey,key)
   nodes <- xml2::xml_find_all(xml, xpath, ns = pkg.env$NAMESPACES)
-	values	<-	sapply(nodes,xml2::xml_text)
+	values	<-	xml2::xml_text(nodes)
```
Comment (wdwatkins) [inline R/geoknife-generic.R]: more tabs here

File: R/query-webdata-variables.R [hunk 2]
```diff
@@ -34,9 +34,8 @@ setMethod(f = "variables_query",signature = c("webdata","webprocess"),
                                        make_listopendapgrids_execute_xml(fabric, knife))
 
             # will error if none found
-            values	<-	sapply(
-              xml2::xml_find_all(gcontent(response),'//gdp:shortname'),
-              xml2::xml_text)
+            values	<-	xml2::xml_text(
```
Comment (wdwatkins) [inline R/query-webdata-variables.R]: and here

File: tests/testthat/test-gconfig.R [hunk 3]
```diff
@@ -22,5 +23,14 @@ test_that("can set !read.only variables", {
 test_that("can set !read.only variables on webprocess()", {
   testthat::skip_on_cran()
   gconfig('email'='none@gmail.com')
-  expect_equal(webprocess()@email, 'none@gmail.com')
+  wp <<-webprocess()
+  expect_equal(wp@email, 'none@gmail.com')
+  expect_is(webprocess(), "webprocess")
+})
+
+context("Test geoknife connection to GDP")
```
[the remark this answers is not in the document]
Comment (dblodgett-usgs) [reply tests/testthat/test-gconfig.R]: Sure, I just moved it to the download_result test. That's the first one that runs against the GDP.

[the remark this answers is not in the document]
Comment (dblodgett-usgs) [reply tests/testthat/test-gconfig.R]: Actually, on second thought, I'll just remove it. I just moved the one thing it's testing uniquely to the test above it.

**Comment by dblodgett-usgs** [comment]
This should be ready to merge.

**Pull request merged**